--- amarok-1.4.10/amarok/src/engine/helix/helix-sp/helix-sp.cpp
+++ amarok-1.4.10/amarok/src/engine/helix/helix-sp/helix-sp.cpp
@@ -1509,7 +1509,7 @@
          delete [] ppctrl[playerIndex]->pszURL;
 
       // see if the file is already in the form of a url
-      char *tmp = strstr(file, "://");
+      const char *tmp = strstr(file, "://");
       if (!tmp)
       {
          char pszURLOrig[MAXPATHLEN];
