diff -urN amarok-1.4.10.orig/amarok/src/actionclasses.cpp amarok-1.4.10/amarok/src/actionclasses.cpp
--- amarok-1.4.10.orig/amarok/src/actionclasses.cpp	2008-08-14 01:21:51.000000000 +0400
+++ amarok-1.4.10/amarok/src/actionclasses.cpp	2011-06-12 16:47:59.273750623 +0400
@@ -30,6 +30,8 @@
 #include <ktoolbarbutton.h>
 #include <kurl.h>
 
+extern KAboutData aboutData;
+
 namespace Amarok
 {
     bool repeatNone() { return AmarokConfig::repeat() == AmarokConfig::EnumRepeat::Off; }
@@ -171,8 +173,6 @@
 KPopupMenu*
 Menu::helpMenu( QWidget *parent ) //STATIC
 {
-    extern KAboutData aboutData;
-
     if ( s_helpMenu == 0 )
         s_helpMenu = new KHelpMenu( parent, &aboutData, Amarok::actionCollection() );
 

