diff --git a/applets/CMakeLists.txt b/applets/CMakeLists.txt
index 00254f6..9fc8b4c 100644
--- a/applets/CMakeLists.txt
+++ b/applets/CMakeLists.txt
@@ -37,12 +37,16 @@ add_subdirectory(notes)
 add_subdirectory(nowplaying)
 if(LIBATTICA_FOUND)
   add_subdirectory(knowledgebase)
-  add_subdirectory(community)
+  if (WITH_KDEWEBKIT)
+    add_subdirectory(community)
+  endif (WITH_KDEWEBKIT)
   add_subdirectory(social-news)
 endif(LIBATTICA_FOUND)
 add_subdirectory(pastebin)
 add_subdirectory(previewer)
-add_subdirectory(rememberthemilk)
+if (WITH_KDEWEBKIT)
+  add_subdirectory(rememberthemilk)
+endif (WITH_KDEWEBKIT)
 add_subdirectory(rssnow)
 add_subdirectory(spellcheck)
 add_subdirectory(showdashboard)
@@ -53,10 +57,10 @@ add_subdirectory(eyes)
 add_subdirectory(unitconverter)
 add_subdirectory(weatherstation)
 
-if(QT_QTWEBKIT_FOUND)
+if(QT_QTWEBKIT_FOUND AND WITH_KDEWEBKIT)
   add_subdirectory(dict)
   add_subdirectory(news)
-endif(QT_QTWEBKIT_FOUND)
+endif(QT_QTWEBKIT_FOUND AND WITH_KDEWEBKIT)
 
 if(PHONON_FOUND)
   add_subdirectory(mediaplayer)
@@ -96,4 +100,6 @@ if (Q_WS_X11)
    endif (X11_Xrender_FOUND)
 endif (Q_WS_X11)
 
-add_subdirectory(webslice)
+if (WITH_KDEWEBKIT)
+  add_subdirectory(webslice)
+endif (WITH_KDEWEBKIT)
diff --git a/dataengines/CMakeLists.txt b/dataengines/CMakeLists.txt
index 15c1aab..4bdfcac 100644
--- a/dataengines/CMakeLists.txt
+++ b/dataengines/CMakeLists.txt
@@ -1,8 +1,8 @@
 add_subdirectory(comic)
 
-if (QCA2_FOUND AND QJSON_FOUND AND QTOAUTH_FOUND)
+if (QCA2_FOUND AND QJSON_FOUND AND QTOAUTH_FOUND AND WITH_KDEWEBKIT)
       add_subdirectory(microblog)
-endif(QCA2_FOUND AND QJSON_FOUND AND QTOAUTH_FOUND)
+endif(QCA2_FOUND AND QJSON_FOUND AND QTOAUTH_FOUND AND WITH_KDEWEBKIT)
 
 if(LIBATTICA_FOUND)
   add_subdirectory(ocs)
diff --git a/dataengines/CMakeLists.txt b/dataengines/CMakeLists.txt
index 15c1aab..4bdfcac 100644
--- a/dataengines/CMakeLists.txt
+++ b/dataengines/CMakeLists.txt
@@ -8,7 +8,9 @@ if(LIBATTICA_FOUND)
 if(LIBATTICA_FOUND)
   add_subdirectory(ocs)
 endif(LIBATTICA_FOUND)

-add_subdirectory(potd)
+if (WITH_KDEWEBKIT)
+  add_subdirectory(potd)
+endif (WITH_KDEWEBKIT)
 add_subdirectory(rememberthemilk)
 add_subdirectory(kdeobservatory)
 add_subdirectory(kimpanel)

diff --git a/libs/rtm/CMakeLists.txt b/libs/rtm/CMakeLists.txt
index 6576082..4fe0488 100644
--- a/libs/rtm/CMakeLists.txt
+++ b/libs/rtm/CMakeLists.txt
@@ -43,4 +43,4 @@ install(TARGETS rtm ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 #install(FILES FindRTM.cmake DESTINATION ${DATA_INSTALL_DIR}/cmake/modules)
 
-add_subdirectory(tests)
+#add_subdirectory(tests)
